src/main/java/TaskList.java

```diff
@@ -0,0 +1,83 @@
+import java.util.ArrayList;
+
+/**
+ * Class to represent all the tasks.
+ * @author vanGoghhh
+ */
+
+public class TaskList {
+
+    private ArrayList<Task> taskList;
+
+    /**
+     * Constructor for tasklist.
+     * @param taskList arraylist containing all the tasks.
+     */
+    public TaskList(ArrayList<Task> taskList) {
+        this.taskList = taskList;
+    }
+
+    /**
+     * Mark a task as completed.
+     * @param completedTask task to be completed.
+     */
+    protected void markTaskDone(Task completedTask) {
+        completedTask.markAsDone();
+        int indexOfTask = this.taskList.indexOf(completedTask);
+        this.taskList.get(indexOfTask).markAsDone();
```

Comment:
Do you need to mark both the task and the item in tasklist as complete? Since it's passed by reference, marking either as done should suffice (unless you cloned the task at some point).